.github/workflows/build-ironic-images.yaml

```
@@ -0,0 +1,35 @@
+---
+name: build-ironic-images
+
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - 'ironic-images/**'
+  push:
+    branches:
+      - main
+    paths:
+      - 'ironic-images/**'
+
+jobs:
+  build-ironic-images:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        id: setup-python
+        with:
+          python-version: '3.11'
+          cache: 'pip'
+      - run: sudo apt update && sudo apt install -y debootstrap qemu-utils
+      - run: pip install -r ironic-images/requirements.txt
+      - run: cd ironic-images/ipa-debian-bookworm && bash ipa-debian-bookworm.sh
+        env:
+          ELEMENTS_PATH: ${{ env.pythonLocation }}/share/ironic-python-agent-builder/dib:${{ github.workspace }}/ironic-images/custom_elements
+      - name: Dynamically set timestamp environment variable
+        run: echo "TIMESTAMP=$(date +"%Y%m%d%H%M%S")" >> $GITHUB_ENV
+      - name: Rename IPA image to include build timestamp
+        run: cd ironic-images/ipa-debian-bookworm/ && cp ipa-debian-bookworm.initramfs ipa-debian-bookworm-${{ env.TIMESTAMP }}.initramfs && cp ipa-debian-bookworm.kernel ipa-debian-bookworm-${{ env.TIMESTAMP }}.kernel
+      - run: cd ironic-images/ipa-debian-bookworm/ && time swift -A https://identity.api.rackspacecloud.com/v1.0 --os-region-name=IAD -U ${{ secrets.RACKSPACE_CLOUD_USERNAME }} -K ${{ secrets.RACKSPACE_CLOUD_API_KEY}} upload --segment-size 5368709120 undercloud-ironic-images ipa-debian-bookworm.initramfs ipa-debian-bookworm.kernel ipa-debian-bookworm-${{ env.TIMESTAMP }}.initramfs ipa-debian-bookworm-${{ env.TIMESTAMP }}.kernel
```

```suggestion
      - name: upload IPA kernel and initramfs
        run: cd ironic-images/ipa-debian-bookworm/ && time swift -A https://identity.api.rackspacecloud.com/v1.0 --os-region-name=IAD upload --segment-size 5368709120 undercloud-ironic-images ipa-debian-bookworm.initramfs ipa-debian-bookworm.kernel ipa-debian-bookworm-${{ env.TIMESTAMP }}.initramfs ipa-debian-bookworm-${{ env.TIMESTAMP }}.kernel
        env:
          ST_USER: ${{ secrets.RACKSPACE_CLOUD_USERNAME }}
          ST_KEY: ${{ secrets.RACKSPACE_CLOUD_API_KEY}}
```

To ensure the secrets are redacted.

---

Updated